### DISCUSSION TITLE
Request for a web interface

### DISCUSSION THREAD

**Discussion by michaelrafailyk** [community/community#143295] [npm] [answered]
Hello. It could be much much easier for most new users to upload their packages using web interface, just like GitHub does.

I'll try to describe my thoughts. I completely understand how it is profitable to use command line to work with packages if you do it daily. But in many cases when I need to just place my simple package of 3 files on npm and forget it for a some time or maybe forever, it could be nice to just upload the files through web interface which doesn't require anything to be installed. But instead, for that simple task, I need to set up complete npm with all dependencies on my computer, like:

- Installing Python. I may understand why, okay, and I already have it for another tasks.
- Installing brew. Oookaay, it's tool for tool.
- Installing nvm. One more tool for tool, but we have fun anyway, so.
- Installing node.js. A whole server to just upload 3 files, really? But I don't need server.
- And finally, install npm.

It's a 5 tools to just upload 3 files. Looks overcomplicated. I already went through this way once when registering the package 4 years ago on my old Mac. I am a technical mind man but I remember how hard that was even for me. Now I have an update for my package and decided to upload its new version to npm. But, I moved to new Mac and realized that now I have to go through this way again. for just upload the package of 3 files. I mean, every time I see that I need another 2-5 applications to be installed to run a simple task that should not require any applications at all, It feels wrong. So please, web interface for npm. Make things easier. Thanks.

**Comment by diata0210** [comment]
You should try sending a message to the United Nations, who knows, it might work!

**Comment by monishcm** [comment, marked as answer]
This is not something we are considering right now, but thanks for your feedback and we will investigate the possibilities in future.

**Reply by michaelrafailyk** [reply]
Hi Monish! Thank you for reply.
